agent/agent.go

```diff
@@ -1254,6 +1260,45 @@ func (a *Agent) clearGitConfig(tc *taskContext) {
 		return
 	}
 	logger.Info("Cleared git credentials.")
+
+	if err := unregisterScalar(); err != nil {
+		logger.Error(errors.Wrap(err, "unregistering scalar repositories"))
```

Comment:
I can't comment on above lines, but would it be better to continue on the errors above? For example, let's say .gitconfig doesn't exist, does that mean ..git-credentials doesn't and we don't need to unregister scalar? I think logging still makes sense and it would need to be rewritten, but I'm thinking in terms of resilience